sensorprocess/lidarsensorprocess.go

```diff
@@ -32,13 +33,18 @@ func (config *Config) addLidarReadingInOnline(ctx context.Context) error {
 	// get next lidar data response; ignoring status since it is always false
 	lidarReading, err := config.Lidar.TimedLidarReading(ctx)
 	if err != nil {
+		if errors.Is(err, replaypcd.ErrEndOfDataset) {
```

Comment:
[note to self] this sleep just to prevent overzealous calling of replay sensor when data will only be synced every sync interval

Reply:
yeah, having the pause here helps reduce the amount of times we call `BinaryDataByFilter` to prevent us from overloading app

Reply:
@JohnN193 do we want this to use as well `strings.Contains(err.Error(), replaypcd.ErrEndOfDataset.Error())`?

Reply:
Was this not already changed in https://github.com/viamrobotics/viam-cartographer/pull/312/files

Reply:
I think only the cases in `sensorprocess.go` were changed - not the ones in `movementsensorprocess.go` and `lidarsensorprocess.go`